### PR TITLE
Add Sleep class that passes all tests

### LINKED ISSUE #27: Create Sleep Class

### PR DESCRIPTION
**Pull request by davidhdaw**
What does this PR do?
-Adds a Sleep class that passes all tests in Sleep-test.js
Where should your reviewer start?
-Line 1 of Sleep.js
What (if any) features are you implementing?
-The Sleep class.
What (if anything) did you refactor?
-Some tests required refactoring in order to pass.
Were there any issues that arose?
-A few of these averages were I think just math errors but now I'm paranoid I'm the one making math errors?
Any other comments, questions, or concerns?
-closes #27

### CODE REVIEW

File: src/Sleep.js [hunk 1]
```diff
@@ -1,5 +1,61 @@
...
+    return this.sleepData.filter(user => user.userID === userID);
+  };
+
+  findAverageSleepHours(userID) {
+    const userData = this.findAllUserData(userID);
+    const averageSleepHours = userData.reduce((acc, dataEntry) => {
+      acc += dataEntry.hoursSlept;
+      return acc
+    }, 0)/userData.length;
+    return Number(averageSleepHours.toFixed(1));
```
Comment (BlueJessen) [inline src/Sleep.js:20]: Nice use of toFixed!

File: src/Sleep.js [hunk 2]
```diff
@@ -1,5 +1,61 @@
...
+    }, 0)/userData.length;
+    return Number(averageSleepQuality.toFixed(1));
+  };
+
+  findDaySleepHours(userID, date) {
+    return this.sleepData.find(userData => userID === userData.userID && date === userData.date).hoursSlept;
+  };
+
+  findDaySleepQuality(userID, date) {
+    return this.sleepData.find(userData => userID === userData.userID && date === userData.date).sleepQuality;
```
Comment (BlueJessen) [inline src/Sleep.js:37]: OOOh I like the && you used here I dont think I knew you could do that. In retrospect it seems obvious but 😤

File: src/Sleep.js [hunk 3]
```diff
@@ -1,5 +1,61 @@
...
+    const dayIndex = userData.findIndex(dataEntry => dataEntry.date === date);
+    const toDate = userData.slice(0 , (dayIndex+1));
+    return toDate.slice(-7).map(date => date.hoursSlept);
+  };
+
+  findWeeklySleepQuality(userID, date) {
+    const userData = this.findAllUserData(userID);
+    const dayIndex = userData.findIndex(dataEntry => dataEntry.date === date);
+    const toDate = userData.slice(0,(dayIndex+1));
+    return toDate.slice(-7).map(date => date.sleepQuality);
```
Comment (BlueJessen) [inline src/Sleep.js:51]: Ohhhh I see what youre saying for the slice now.

**Review by BlueJessen — APPROVED** [review]

**Review by LINTONBNEIL — APPROVED** [review]
Really nice work setting up everything for the sleep class, we've all done an amazing job throughout this project.

**Pull request merged**